pvlib/tests/test_modelchain.py

```diff
@@ -327,6 +334,88 @@ def test_run_model_tracker(sapm_dc_snl_ac_system, location, weather, mocker):
     assert np.isnan(mc.ac[1])
 
 
+def test__assign_total_irrad(sapm_dc_snl_ac_system, location, weather,
+                             total_irrad):
+    weather[['poa_global', 'poa_diffuse', 'poa_direct']] = total_irrad
+    mc = ModelChain(sapm_dc_snl_ac_system, location)
+    mc._assign_total_irrad(weather)
+    for k in modelchain.POA_DATA_KEYS:
+        assert_series_equal(mc.total_irrad[k], total_irrad[k])
+
+
+def test_prepare_inputs_from_poa(sapm_dc_snl_ac_system, location,
+                                 weather, total_irrad):
+    data = weather.copy()
+    data[['poa_global', 'poa_diffuse', 'poa_direct']] = total_irrad
+    mc = ModelChain(sapm_dc_snl_ac_system, location)
+    mc.prepare_inputs_from_poa(data)
+    # weather attribute
+    assert_frame_equal(mc.weather, weather)
```

Comment:
@cwhanse this `assert_frame_equal` fails when I run the tests locally but it seems to pass on the CI. Looking at the code, I expect that it should fail because `ModelChain._assign_weather` adds `wind_speed` and `temp_air` to `mc.weather`. Is my understanding correct? If so, why don't we see test failures on the CI?

Reply:
Odd, that tested OK locally for me. Your understanding of `_assign_weather` is correct. I question my understanding of `pytest.fixture`

Reply:
https://github.com/pvlib/pvlib-python/pull/1052